### PR TITLE
Add hl7v2 to FHIR guide

### PR DESCRIPTION
**Pull request by isuruuy429**
This PR contains the following. 
- The Ballerina example of hl7v2 to FHIR transformation
- The Ballerina example of customization v2 segments.

### CODE REVIEW

File: en/docs/data-transformation/guides/hl7v2-fhir.md [hunk 1]
```diff
@@ -0,0 +1,364 @@
+## Introduction
+The hl7v2 to fhirr4 library is a Ballerina module designed to facilitate the conversion of healthcare data from the HL7 version 2 (HL7v2) standard to the FHIR R4 standard. In the healthcare industry, interoperability and data exchange between different systems are critical for ensuring accurate and efficient patient care. This library provides developers with a straightforward way to transform HL7v2 messages into FHIR R4 resources, enabling seamless integration with modern healthcare systems that utilize the FHIR standard.
```
Comment (sameeragunarathne) [inline en/docs/data-transformation/guides/hl7v2-fhir.md]: Let's not specify about ballerina in the introduction, later we will have the seperate ballerina tab

File: en/docs/data-transformation/guides/hl7v2-fhir.md [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+
+## Mapping ADT_A01 Message to FHIR r4
+In hl7v2, there are Messages, Segments and Data types that we can map to FHIR. In this example let's see how we can map ADT_A01.PID segment to FHIR Patient. 
+
```
Comment (sameeragunarathne) [inline en/docs/data-transformation/guides/hl7v2-fhir.md:6]: Let's add the note that we have two implementation flavours

{!includes/bal-mi-note.md!}

File: en/docs/data-transformation/guides/hl7v2-fhir.md (hunk 1 above)
Comment (isuruuy429) [reply en/docs/data-transformation/guides/hl7v2-fhir.md]: This is fixed.

File: en/docs/data-transformation/guides/hl7v2-fhir.md (hunk 2 above)
Comment (isuruuy429) [reply en/docs/data-transformation/guides/hl7v2-fhir.md:6]: This is fixed.

**Review by sameeragunarathne — APPROVED** [review]

**Pull request merged**